libraries/WiFi/src/WiFiMulti.cpp

```diff
@@ -199,6 +266,143 @@ uint8_t WiFiMulti::run(uint32_t connectTimeout)
         // scan wifi async mode
         WiFi.scanNetworks(true);
     }
-
     return status;
 }
+
+void WiFiMulti::markAsFailed(int32_t i) {
+    APlist[i].fail = true;
+    log_i("marked %s as failed",APlist[i].ssid);
+}
+
+void WiFiMulti::resetFails(){
+    log_i("resetting failure flags");
+    for(uint32_t i = 0; i < APlist.size(); i++) {
+        APlist[i].fail = false;
+    }
+}
+
+void WiFiMulti::setStrictMode(bool bStrict) {
+    _bStrict = bStrict;
+}
+
+void WiFiMulti::setAllowOpenAP(bool bAllowOpenAP) {
+    _bAllowOpenAP = bAllowOpenAP;
+}
+
+void WiFiMulti::setTestConnection(bool bTestConnection){
+    _bTestConnection = bTestConnection;
+}
+void WiFiMulti::setTestPhrase(const char* testPhrase){
+    _testPhrase = testPhrase;
+}
+void WiFiMulti::setTestURL(String testURL){
+    _testURL = testURL;
+}
+
+bool WiFiMulti::testConnection(){
```

Comment:
why not add the URL as an argument (defaulting to NULL) and then parse it to get the port and so on?

Reply:
I figured most people would just want to turn it on and forget it so I defaulted the test URL and phrase to amazon which I thought would be universal.